lib/vagrant-openshift/action/build_sti_base_windows.rb

```diff
@@ -0,0 +1,128 @@
+          raise "AWS credentials not configured" if aws_creds.nil?
+          compute = Fog::Compute.new(Vagrant::Openshift::Aws::fog_config(aws_creds))
+
+          @env[:ui].info("Finding base AMI...")
+          images = compute.images.all({"Owner" => "amazon", "name" => "Windows_Server-2012-R2_RTM-English-64Bit-Base-*", "state" => "available"})
+          # https://github.com/fog/fog-aws/issues/320 would make this less hacky
+          images.sort_by! {|image| image.name[-10,10]}
+          @env[:ui].info("Using base AMI #{images.last.name}, #{images.last.id}")
+
+          @env[:ui].info("Creating instance...")
+          instance = compute.servers.create(
+            :tags => {"Name" => "#{@options[:instance_prefix]}-windows2012r2_#{Time.now.to_i}"},
+            :image_id => images.last.id,
+            :flavor_id => "m4.large",
+            :key_name => "libra",
+            :subnet_id => "subnet-cf57c596",
+            :user_data => <<'USERDATA'
+<powershell>
+Set-ExecutionPolicy -ExecutionPolicy Bypass
+
+# Download and install Cygwin
+$client = New-Object System.Net.WebClient
+$client.DownloadFile("https://cygwin.com/setup-x86_64.exe", "C:\Windows\Temp\setup-x86_64.exe")
+&C:\Windows\Temp\setup-x86_64.exe -q -D -L -d -o -s http://mirrors.kernel.org/sourceware/cygwin -l C:\Windows\Temp\cygwin -R C:\cygwin -P curl -P gcc-core -P git -P make -P openssh -P rsync | Out-Null
+Remove-Item -Recurse C:\Windows\Temp\cygwin
+Remove-Item C:\Windows\Temp\setup-x86_64.exe
+$env:Path += ";C:\cygwin\bin"
+
+# Configure and start OpenSSH
+$env:LOGONSERVER = "\\" + $env:COMPUTERNAME  # http://petemoore.github.io/general/taskcluster/2016/03/30/windows-sshd-cygwin-ec2-aws.html
+&bash -c "ssh-host-config --yes -w Pa$$w0rd"
+&bash -c "sed -i -e 's/#PasswordAuthentication yes/PasswordAuthentication no/' /etc/sshd_config"
+&bash -c "sed -i -e 's/#ChallengeResponseAuthentication yes/ChallengeResponseAuthentication no/' /etc/sshd_config"
+&cygrunsrv -S sshd
+&netsh advfirewall firewall add rule name=SSH profile=any localport=22 enable=yes action=allow dir=in protocol=tcp
+
+# Download and install Golang
+$client.DownloadFile("https://storage.googleapis.com/golang/go1.7.1.windows-amd64.msi", "C:\Windows\Temp\go1.7.1.windows-amd64.msi")
+&msiexec /qb /i C:\Windows\Temp\go1.7.1.windows-amd64.msi | Out-Null
+Remove-Item C:\Windows\Temp\go1.7.1.windows-amd64.msi
+
+# Create Administrator's home directory
+Out-Null | &bash --login
+&bash -c "mkdir /home/Administrator/.ssh; curl -o /home/Administrator/.ssh/authorized_keys http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key"
+&bash -c "mkdir /data"
+$stream = New-Object System.IO.StreamWriter @("c:\cygwin\home\Administrator\.bash_profile", $true)
+$stream.Write("export GOPATH='C:\cygwin\data'`n")
+$stream.Close()
+
+# Download and install Docker client
+$client.DownloadFile("https://get.docker.com/builds/Windows/x86_64/docker-1.10.3.exe", "C:\cygwin\usr\local\bin\docker.exe")
```

Comment:
I don't regard it as a problem and I'm not aware of any obvious alternatives anyway.